pygradle-plugin/src/test/groovy/com/linkedin/gradle/python/extension/PythonDetailsTest.groovy

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2016 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.gradle.python.extension
+
+import org.gradle.api.GradleException
+import org.gradle.testfixtures.ProjectBuilder
+import spock.lang.Specification
+
+class PythonDetailsTest extends Specification {
+
+    def project = new ProjectBuilder().build()
+    def details = new PythonDetails(project)
+    def originalWhitelistedPythonVersions = PythonVersion.whitelistedPythonVersions
+    def originalDefaultPython2 = PythonVersion.defaultPython2
+    def originalDefaultPython3 = PythonVersion.defaultPython3
+
+    def cleanup() {
+        PythonVersion.whitelistedPythonVersions = originalWhitelistedPythonVersions
+        PythonVersion.defaultPython2 = originalDefaultPython2
+        PythonVersion.defaultPython3 = originalDefaultPython3
+    }
+
+    def 'test normalize to acceptable version 3.x'() {
```

Comment:
Take a look at Spock's `Data Tables` (http://spockframework.org/spock/docs/1.1/data_driven_testing.html) . You could collapse a lot of these into one test with an `@Unroll` on the method.